File: Algorithms/src/693. Binary Number with Alternating Bits/Code/solution.py
class Solution:
    def hasAlternatingBits(self, n: int) -> bool:

        prev = n & 1
        n >>= 1

        while n > 0:
            curr = n & 1

            if curr == prev:
                return False

            prev = curr
            n >>= 1

        return True

File: Algorithms/src/693. Binary Number with Alternating Bits/Code/solution.java
class Solution {
    public boolean hasAlternatingBits(int n) {

        int prev = n & 1;
        n >>= 1;

        while (n > 0) {
            int curr = n & 1;

            if (curr == prev) {
                return false;
            }

            prev = curr;
            n >>= 1;
        }

        return true;
    }
}
